mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,262 @@
+
+This section will explain how to create, activate, use and deactivate Python virtual environments.
+
+### Creating a Python virtual environment
+
+A Python virtual environment can be created with the following command:
+
+```bash
+$ python -m venv myenv      # Create a new virtual environment named myenv
+```
+
+This command creates a new directory named `myenv` in the current working directory. 
+This directory will contain the packages, scripts, and binaries that are needed to manage the virtual environment.
+
+### Activating a virtual environment
+
+To use the virtual environment, you need to activate it. 
+This will modify the shell environment to use the Python interpreter and packages from the virtual environment.
+
+```bash
+$ source myenv/bin/activate # Activate the virtual environment
+```
+
+### Installing packages in a virtual environment
+
+After activating the virtual environment, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+These packages will be scoped to the virtual environment and will not affect the system-wide Python installation, 
+and are only available when the virtual environment is activated.
+No administrator privileges are required to install packages in a virtual environment.
+
+It is now possible to run Python scripts that use the installed packages in the virtual environment.
+
+!!! note
+    Always prefer to use Python packages that are centrally installed, which are available via the environment modules interface.
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
+    ```
+
+    Some Python packages are installed as extensions of modules. For example, `numpy`, `scipy` and `pandas` are 
+    part of the `SciPy-bundle` module. You can use
+
+    ```bash
+    $ module show <module_name>
+    ```
+
+    to check which extensions are included in a module, if any.
+
+
+### Deactivating a virtual environment
+
+When you are done using the virtual environment, you can deactivate it.
+To do that, run:
+
+```bash
+$ deactivate
+```
+
+## Combining virtual environments with centrally installed modules
+
+You can combine Python packages installed in a virtual environment with environment modules. 
+The following script uses PyTorch (which is available as a module)
+and Poutyne (which we assume is not centrally installed):
+
+```python title="pytorch_poutyne.py"
+import torch
+import poutyne
+
+...
+```
+
+We load a PyTorch package as a module and install Poutyne in a virtual environment:
+
+```bash
+$ module load PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+$ pip install Poutyne
+```
+
+While the virtual environment is activated, we can run the script without any issues:
+
+```bash
+$ python pytorch_poutyne.py
```

Comment:
activate first

Reply:
The environment is activated in the previous codeblock

Reply:
Yes, but as mentioned elsewhere, I would things apart in an "install" subsection, and a "use" subsection.

And to avoid confusion, it's better to use "`deactivate`" after installing the virtual environment, and `activate` again right before using it (and also load the necessary modules first)